docs/proposals/job_controller.md

```diff
@@ -0,0 +1,192 @@
+    // PendingExecutions are the job runs pending for execution
+    PendingExecutions []JobExec
+
+    // CurrentExecutions are the job run currently executing
+    RunningExecutions []JobExec
+
+    // CompletedExecutions tracks previously scheduled jobs (up to a limited amount)
+    CompletedExecutions []JobExec
+
+    // ScheduleCount tracks the increasing amount of executions for this job
+    ScheduleCount int
+}
+
+// JobExec represents the current state of a single execution of a Job.
+type JobExec struct {
+    // CreatedAt represents time when the job execution was created
+    CreatedAt util.Time
+
+    // StartedAt represents time when the job execution was started
+    StartedAt util.Time
+
+    // CompletedAt represents time when the job execution was completed
+    CompletedAt util.Time
+
+    // Phase represents the point in the job execution lifecycle.
+    Phase JobExecPhase
+
+    // Tag is added in labels of pod(s) created for this job execution.  It allows
+    // job object to safely group/track all pods started for one given job execution.
+    Tag util.UID
+}
+
+// JobExecPhase represents job execution phase at given point in time.
+type JobExecPhase string
+
+// These are valid JobExec phases.
+const (
+    // JobExecPending means the pod has been accepted by the system but one or more
+    // pods has not been started.
+    JobExecPending JobExecPhase   = "Pending"
+    // JobExecRunning means that all pods have been started.
+    JobExecRunning JobExecPhase   = "Running"
+    // JobExecComplete means that all pods have terminated with an exit code of 0.
+    JobExecComplete JobExecPhase = "Complete"
+    // JobExecFailed means that all pods have terminated and at least one of
+    // them terminated with non-zero exit code.
+    JobExecFailed JobExecPhase = "Failed"
+)
+```
+
+## JobSpec
+`JobSpec` structure is defined to contain all information how the actual job execution
+will look like, including the schedule in ISO8601 format, pod specification, etc.
+
+```
+// JobSpec describes how the job execution will look like and when it will actually run.
+type JobSpec struct {
+
+    // ScheduleSpec contains the schedule in ISO8601 format.
+    ScheduleSpec string
+
+    // SkipOutdated specifies that if AllowConcurrent is false, only the newest job
+    // will be started (default: true), ignoring executions that missed their schedule.
+    SkipOutdated bool
+
+    // BlockOnFailure suspends scheduling of next job runs after a failed one.
+    BlockOnFailure bool
+
+    // AllowConcurrent specified whether concurrent jobs may be started.
+    AllowConcurrent bool
+
+    // PodCount specifies the number of pods to start for each job run.
+    PodCount int
+
+    // Selector is a label query over pods that should match the pod count.
+    Selector map[string]string
+
+
+    // Template is the object that describes the pod that will be created when
+    // executing a job.
+    Template *PodTemplateSpec
```

Comment:
Are you planning to specify metadata here, as well as a `PodSpec`?  If not, all you need is a `PodSpec`.